### DISCUSSION TITLE
can't import module from cdn

### DISCUSSION THREAD

**Discussion by jimmywarting** [alpinejs/alpine#3787] [7. Bugs (archived)]
<img width="989" alt="image" src="https://github.com/alpinejs/alpine/assets/1148376/10946c51-989f-41f5-a068-5b16b26c4c8b">

that's b/c you use extension-less path in esm. 

you should change it to:

```diff
- import Alpine from './../src/index'
+ import Alpine from './../src/index.js'

export default Alpine
```

**Comment by ekwoka** [comment]
use an esm host like esm.sh

```js
import Alpine from 'https://esm.sh/alpinejs'
```

**Reply by jimmywarting** [reply]
Still think that you should use explicit path in inports. 
- ESM are strict about it. 
- You would barely need any build tool at all to compile your own code in order to test it.
  - then it would only be a bundler / minifier
- ppl can try out the original, unminified source code that have better debugging experience and use the bundled one for production
- looking up file on the disk gets faster cuz it dose not have to 2nd guess what it meant to import by having to scan the filesystem for potential files that you need / want to import

**Reply by SimoTod** [reply]
The repo is open source. You have better chances to see that implemented if you submit a PR explaining the benefits.

**Reply by ekwoka** [reply]
Regardless, the src won't work like that, because the verison is only referenced as ALPINE_VERSION

which is defined by esbuild during bundle
